### DISCUSSION TITLE
how to inference with tensorrt on rectangular images?

### DISCUSSION THREAD

**Discussion by wookidooki8** [ultralytics/yolov5#7913] [Q&A]
i can export to tensorrt for square image size such as 640x640 or 512x512 but
how do i export it for rectangular image size for example 640x320?  
(--dynamic doesnt seem to work and giving specific img size using --imgsz seems to work with square img size only.)

**Comment by glenn-jocher** [comment]
@wookidooki8 try looking at the argparser. `python export.py --img 640 320`

**Reply by wookidooki8** [reply]
@glenn-jocher thankx. the command you gave worked but im getting this error when inferencing. (using torch hub)
   
    test = img.jpg #480x320
    model = torch.hub.load('./', 'custom', path='best.pt', source='local', force_reload = True)
    results = model(test, size = 480)

    AssertionError: (torch.Size([1, 3, 480, 480]), (1, 3, 480, 320)) 

is there any command for using rect inference when using torch.hub?

**Reply by glenn-jocher** [reply]
@wookidooki8 you may need to supply your own pre and post processing, or else submit a PR to update hub rectangular inference for exported formats

**Reply by glenn-jocher** [reply]
detect.py might also be an option, but right now rectangular inference for exported formats is a bit experimental.

**Reply by CourchesneA** [reply]
@wookidooki8  did you make any progress on this issue ? We are trying to achieve the same thing

**Reply by abdullahabid10** [reply]
@glenn-jocher any updates on this? Will rectangular inference using detect.py work for tensorrt exported model?

**Reply by glenn-jocher** [reply]
@abdullahabid10 yes this should work normally, just export and run inference at the same rectangular image size, i.e. `--imgsz 640 1280`

**Reply by abdullahabid10** [reply]
@glenn-jocher thanks for responding. Let me clarify my question: if I export to tensorrt with `--imgsz 640 640`, will rectangular inference work by default like it does for .pt model (https://github.com/ultralytics/yolov3/issues/232) if I use detect.py?

**Reply by glenn-jocher** [reply]
You can use the --dynamic flag during TRT export if you want dynamic image size inference.
